Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -387,7 +391,9 @@ WebIDL::ExceptionOr<void> HTMLInputElement::set_value(String const& value)
     auto old_value = move(m_value);
 
     // 2. Set the element's value to the new value.
-    // NOTE: This is done as part of step 4 below.
+    // NOTE: For the TextNode this is done as part of step 4 below.
+    if (type_state() == TypeAttributeState::Color)
+        MUST(m_color_well_element->style_for_bindings()->set_property(CSS::PropertyID::BackgroundColor, m_value));
```

Comment:
You might have trouble with this. It's possible for JS to set the value before the element is added to the document, so `m_color_well_element` could be null. So any place like this, you need to also check that it's not null.